.devcontainer/Dockerfile

```
@@ -97,12 +97,15 @@ RUN echo "alias ll='ls -lah'" >> /home/$USERNAME/.bashrc \
     && echo "alias gtr='go test -race ./...'" >> /home/$USERNAME/.bashrc
 
-# Ensure ~/.local/bin is on PATH for bash (Claude Code, etc.)
-RUN echo 'export PATH="$HOME/.local/bin:$PATH"' >> /home/$USERNAME/.bashrc
+# Ensure ~/.local/bin and ~/.opencode/bin are on PATH for bash
+RUN echo 'export PATH="$HOME/.local/bin:$HOME/.opencode/bin:$PATH"' >> /home/$USERNAME/.bashrc
 
 # Enable bash completion
 RUN echo "source /etc/bash_completion" >> /home/$USERNAME/.bashrc
 
+# Install OpenCode AI coding agent
+RUN curl -fsSL https://opencode.ai/install | bash
```

Hadolint (2.14.0)

[warning] Set the SHELL option -o pipefail before RUN with a pipe in it. If you are using /bin/sh in an alpine image or if your shell is symlinked to busybox then consider explicitly setting your SHELL to /bin/ash, or disable this check

(DL4006)